src/net.c

```diff
@@ -130,11 +130,30 @@ char *iptostr(struct sockaddr *sa)
  */
 int setsockname(sockname_t *addr, char *src, int port, int allowres)
 {
+  char *endptr;
+  long val;
+  IP ip;
   struct hostent *hp;
   int af = AF_UNSPEC;
 #ifdef IPV6
+  char ip2[INET6_ADDRSTRLEN];
   int pref;
+#else
+  char ip2[INET_ADDRSTRLEN];
+  int i, count;
+#endif
 
+  /* DCC CHAT ip is expressed as integer but inet_pton() only accepts dotted
+   * addresses */
+  val = strtol(src, &endptr, 10);
+  if (val && !*endptr) {
+    ip = htonl(val);
+    if (inet_ntop(AF_INET, &ip, ip2, sizeof ip2)) {
```

Comment:
inet_ntop takes struct in_addr *, but &ip is IP*?

Reply:
[...] The function inet_ntop() converts an address *src from network format (usually a struct in_addr **or some other binary form** [...]

See: https://www.freebsd.org/cgi/man.cgi?query=inet_ntop&apropos=0&sektion=0&manpath=FreeBSD+12.0-RELEASE+and+Ports&arch=default&format=html